addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CookieLooselyScopedScanRule.java

```diff
@@ -76,88 +70,50 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
             }
         }
 
-        // raise alert if have found any loosely scoped cookies
-        if (looselyScopedCookies.size() > 0) {
+        // raise an alert if any loosely scoped cookies were found
+        if (!looselyScopedCookies.isEmpty()) {
             buildAlert(host, looselyScopedCookies).raise();
         }
     }
 
     /*
      * Determines whether the specified cookie is loosely scoped by
-     * checking it's Domain attribute value against the host
+     * checking its Domain attribute value against the origin domain.
      */
-    private static boolean isLooselyScopedCookie(HttpCookie cookie, String host) {
+    private static boolean isLooselyScopedCookie(HttpCookie cookie, String originDomain) {
         // preconditions
         assert cookie != null;
-        assert host != null;
+        assert originDomain != null;
 
         String cookieDomain = cookie.getDomain();
 
-        // if Domain attribute hasn't been specified, the cookie
-        // is scoped with the response host
+        // No problem here since by default the cookie is
+        // scoped to the origin, not including subdomains
         if (cookieDomain == null || cookieDomain.isEmpty()) {
             return false;
         }
 
-        // Split cookie domain into sub-domains
-        String[] cookieDomains = cookie.getDomain().split("\\.");
-        // Split host FQDN into sub-domains
-        String[] hostDomains = host.split("\\.");
-
-        boolean isFromTheSameDomain = isCookieAndHostHaveTheSameDomain(cookieDomains, hostDomains);
-        if (!isFromTheSameDomain) {
-            return true;
-        }
-        // if cookie domain doesn't start with '.', and the domain is
-        // not a second-level domain (example.com), the cookie Domain and
-        // host values should match exactly
-        if (!cookieDomain.startsWith(".") && cookieDomains.length >= 2 && !isFromTheSameDomain) {
-            return !cookieDomain.equals(host);
-        }
+        cookieDomain = cookieDomain.toLowerCase();
+        originDomain = originDomain.toLowerCase();
 
-        // otherwise, remove the '.' and compare the result with the host
-        if (cookieDomains.length != 2) {
-            cookieDomains = cookieDomain.substring(1).split("\\.");
+        // If the cookie domain starts with a leading dot,
+        // remove it as per RFC (ignore the dot)
```

Comment:
Would be good to state somewhere which RFC is specifically since "latest" changes over time.